ntoskrnl/ps/process.c

```diff
@@ -712,8 +714,26 @@ PspCreateProcess(OUT PHANDLE ProcessHandle,
     /* Check if the parent had a job */
     if ((Parent) && (Parent->Job))
     {
-        /* FIXME: We need to insert this process */
-        DPRINT1("Jobs not yet supported\n");
+        Job = Parent->Job;
+
+        if (!(Job->LimitFlags & JOB_OBJECT_LIMIT_SILENT_BREAKAWAY_OK))
+        {
+            /* Requested to create it without a job */
+            if (Flags & PROCESS_CREATE_FLAGS_BREAKAWAY)
+            {
+                /* Is this allowed? */
+                if (!(Job->LimitFlags & JOB_OBJECT_LIMIT_BREAKAWAY_OK))
+                {
+                    Status = STATUS_ACCESS_DENIED;
+                    goto CleanupWithRef;
+                }
+            }
+            else if (InterlockedCompareExchangePointer((PVOID)&Process->Job, Parent->Job, NULL) == NULL)
```

Comment:
Is the Job member only protected by interlocking? If so, we're not retrieving Parent->Job correctly.
Also, should `Parent->Job` be the local `Job` variable instead, here and below?